### PR TITLE
[MRG] Include 'part' in the entity list

### PR DESCRIPTION
**Pull request by teonbrooks**
Update utils.py

Thanks for contributing. If this is your first time,
make sure to read [contributing.md](https://github.com/mne-tools/mne-bids/blob/master/CONTRIBUTING.md)

PR Description
--------------

`part` is missing from the entity list. It is introduced for the fif files in the appendix but it isn't being captured in our parsing step

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by teonbrooks** [comment]
`part` is missing from the entity list. It is introduced for the fif files in the appendix but it isn't being captured in our parsing step

**Comment by jasmainak** [comment]
test please :-)

File: mne_bids/tests/test_utils.py [hunk 1]
```diff
@@ -219,10 +219,10 @@ def test_parse_ext():
 
 
 @pytest.mark.parametrize('fname', [
-    'sub-01_ses-02_task-test_run-3_meg.fif',
-    'sub-01_ses-02_task-test_run-3.fif',
-    'sub-01_ses-02_task-test_run-3',
-    '/bids_root/sub-01/ses-02/meg/sub-01_ses-02_task-test_run-3_meg.fif',
+    'sub-01_ses-02_task-test_run-3_part-01_meg.fif',
+    'sub-01_ses-02_task-test_run-3_part-01.fif',
+    'sub-01_ses-02_task-test_run-3_part-01',
```
Comment (teonbrooks) [inline mne_bids/tests/test_utils.py]: @jasmainak, do you know why this and the line above it is being tested. neither are valid bids-compatible filenames

Comment (jasmainak) [reply mne_bids/tests/test_utils.py]: to allow for making bids_basename?

**Comment by teonbrooks** [comment]
@jasmainak it's all good to go

**Comment by jasmainak** [comment]
do you want to update `whats_new.rst`? otherwise, it's good to go

**Comment by jasmainak** [comment]
sorry there are merge conflicts now ...

**Comment by jasmainak** [comment]
Thanks @teonbrooks !

**Pull request merged**